src/cpp/include/openvino/genai/generation_config.hpp

```diff
@@ -78,7 +78,8 @@ class OPENVINO_GENAI_EXPORTS GenerationConfig {
     size_t max_length = SIZE_MAX;
     bool ignore_eos = false;
     size_t min_new_tokens = 0;
-
+    bool echo = true; //TODO: switch to false
```

Comment:
Echo usually repeats exactly after a user, but there's nothing to repeat after except for the probmpt. This looks more like debug logging. While adding a function for debugging is debatable in https://github.com/openvinotoolkit/openvino.genai/pull/677#discussion_r1722008733, your case is debug logging. It must be handled with compile time definition or env var

Reply:
I'd argue that it's not just debug logging. `echo` is a regular parameter in OpenAI completions API:
https://platform.openai.com/docs/api-reference/completions/create#completions-create-echo
It repeats the prompt, but also processes it - so it can be valuable to analyze the prompt. When `echo` is set, results for the prompt are sent along with the completion (unless we set max_new_tokens == 0, then only prompt is returned since there is no completion).

vLLM supports it as well:
https://github.com/vllm-project/vllm/pull/1504/files

Reply:
Ok.

Please, add dockstrings and python bindings

Reply:
Sure, I'll still have to work on the overall correctness, then I'll add it + tests for this feature as well.